runtime/near-vm-runner/src/cache.rs

```diff
@@ -326,8 +332,14 @@ pub mod wasmer2_cache {
         config: &VMConfig,
         cache: Option<&dyn CompiledContractCache>,
     ) -> Result<Result<VMArtifact, CompilationError>, CacheError> {
+        let vm = Wasmer2VM::new(config.clone());
         match cache {
-            None => Ok(compile_module_wasmer2(code.code(), config)),
+            None => Ok(compile_module_wasmer2(&vm, code.code(), config).and_then(|executable| {
+                vm.engine
+                    .load_universal_executable(&executable)
```

Comment:
Hm, what kind of error can we get during loading? I'd assume that this step can only fail catastrophically (mmap fails, or something), and that we should just panic here. But I assume the reality doesn't work like that? :)

Reply:
Failure modes here are indeed representing mostly internal failure modes – function body being longer than 32-bits, failure to decode a data initializer, allocate data and such.

Reply:
Hm, "failure to allocate data" in particular seems like something non-deterministic (can happen on one machine, but not the other), and that's a big no-no for us. This feels like something we should fix the sooner the better. Although, if this doesn't make the situation worse (ie, we already were treating allocation errors as compilation/loading errors), I think it's fine to land this.

Reply:
Yeah, I've been thinking about this a lot lately. Mostly in context of also making things faster with some arenas ^^